### DISCUSSION TITLE
[GNU/Linux -> Android] Download fails/doesn't start.

### DISCUSSION THREAD

**Discussion by ghost** [claudiodangelis/qr-filetransfer#340] [Q&A] [answered]
When sending files to my Android device, it works fine on full-featured browsers like Firefox, Chrome, Brave, etc. but on lightweight Android WebView implementations like the DuckDuckGo Browser (my browser of choice), Via Browser, XBrowser, Incognito Browser, etc, the download fails/doesn't start. Are there features missing from those lightweight browsers that qrcp needs (if so, what could it be?). Is there any way I can get it to work on those browsers? I can send files from Android to my PC just fine though.
P.S. thanks for making this amazing tool that I use almost everyday!

**Comment by claudiodangelis** [comment]
Thanks for reporting! I have no idea what the root cause would be, please allow me some time for investigating. 

Thanks,
C

**Reply by ghost** [reply, marked as answer]
I have some updates.
I was successful in downloading through the DuckDuckGo browser (and by extension, any other similar browsers) by simply enabling the --keep-alive option ($ qrcp send --keep-alive {file}). I guess my issue can be considered solved, but I still think it's peculiar that the --keep-alive option is needed only for the lightweight android browsers. So I still think it's worth investigating.
